juju/juju
# Synchronise DB Upgrade Across Controllers

## Description of change

This patch introduces logic so that on controllers not running the primary MongoDB, will wait for the primary DB upgrade steps to have run before unlocking the DB upgrade gate.

The same document used by the `upgradesteps` worker is used for coordination, with an additional status to represent the database upgrade having been run.

## QA steps

Simple current-version deployment.
- Bootstrap and enable HA.
- Deploy some units.
- Observe there are no associated errors in the logs.

Upgrade with HA.
- Bootstrap the 2.6 branch and enable HA.
- Deploy some units.
- Build this patch and upgrade the controller.

Upgrade on CAAS.
- Bootstrap 2.6 on Microk8s.
- Deploy something such as `juju deploy cs:~kubeflow-charmers/mariadb-k8s-7`.
- Build this patch, update the operator image, and upgrade the controller.

## Documentation changes

None.

## Bug reference

N/A
